controllers/bootstrap/init.go

```diff
@@ -1758,3 +1771,23 @@ func (b *Bootstrap) DeployCertManagerCR() error {
 	}
 	return nil
 }
+
+// delete cs certmanager resource in v1alpha1 version
+func (b *Bootstrap) deletev1alpha1Resource(kind string, name string) error {
+	klog.Infof("Deleting kind: %s name: %s in namespace: %s", kind, name, b.CSData.MasterNs)
+	v1alpha1Resource := util.NewUnstructured("certmanager.k8s.io", kind, "v1alpha1")
+	if err := b.Client.Get(ctx, types.NamespacedName{Name: name, Namespace: b.CSData.MasterNs}, v1alpha1Resource); err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+		klog.V(2).Infof("There is no v1alpha1 cs-ca-certificate in the cluster")
+	}
+
+	if err := b.Client.Delete(context.TODO(), v1alpha1Resource); err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
Hi @YuchenLi223 

May I ask that if we could have a more generic function for deleting resources based on Group, Version and Kind?
We could refer to the existing function here to see if we could combine them together.
https://github.com/IBM/ibm-common-service-operator/blob/c1ee67b6c687d2a925523618a1a87dfbb26bbefb/controllers/goroutines/cleanup_deprecated_services.go#L229

Reply:
Ah I didn't find it, will use `cleanup` function now